velox/exec/tests/RowContainerTest.cpp

```diff
@@ -284,121 +285,304 @@ class RowContainerTest : public exec::test::RowContainerTestBase {
     return rowContainer;
   }
 
+  // Count number of nulls at the start of the list.
   template <typename T>
-  void testCompareFloats(TypePtr type, bool ascending, bool nullsFirst) {
-    // NOTE: set 'isJoinBuild' to true to enable nullable sort key in test.
-    auto rowContainer = makeRowContainer({type}, {type}, false);
-    auto lowest = std::numeric_limits<T>::lowest();
-    auto min = std::numeric_limits<T>::min();
-    auto max = std::numeric_limits<T>::max();
-    auto nan = std::numeric_limits<T>::quiet_NaN();
-    auto inf = std::numeric_limits<T>::infinity();
-
-    facebook::velox::test::VectorMaker vectorMaker{pool_.get()};
-    VectorPtr values = vectorMaker.flatVectorNullable<T>(
-        {std::nullopt, max, inf, 0.0, nan, lowest, min});
-    int numRows = values->size();
-
-    SelectivityVector allRows(numRows);
-    DecodedVector decoded(*values, allRows);
-    std::vector<char*> rows(numRows);
-    std::vector<std::pair<int, char*>> indexedRows(numRows);
-    for (size_t i = 0; i < numRows; ++i) {
-      rows[i] = rowContainer->newRow();
-      rowContainer->store(decoded, i, rows[i], 0);
-      indexedRows[i] = std::make_pair(i, rows[i]);
+  static size_t countLeadingNulls(const std::vector<std::optional<T>>& values) {
+    int32_t numNulls{0};
+    int32_t i{0};
+    while (i < values.size() && !values[i].has_value()) {
```

Comment:
nit: perhaps, this can be simplified using a for loop

```
for (auto i = 0 ; i < values.size(); ++i) {
   if (values[i].has_value()) {
      return i;
   }
}

return values.size();
```